sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/Uri.java

```diff
@@ -230,12 +249,31 @@ public int getPriority() {
     }
 
     static class HealthStatusAndDiagnosticStringTuple {
-        private final String diagnsoticString;
         private final HealthStatus status;
+        private final String ATTEMPTING = "Attempting";
+        private final String NOT_ATTEMPTING = "NotAttempting";
+        private final String PRIMARY = "Primary";
+        private final String SECONDARY = "Secondary";
+        private final URI uri;
+        private boolean isAttempting;
+        private boolean isPrimary;
 
         public HealthStatusAndDiagnosticStringTuple(URI uri, HealthStatus status) {
-            this.diagnsoticString = uri.getPort() + ":" + status;
+            this.uri = uri;
             this.status = status;
         }
+
+        public String getDiagnsoticString() {
+            return uri.getPort() + ":" + (isPrimary ? PRIMARY : SECONDARY) + ":" +
```

Comment:
I would personally prefer categorizing attempting vs. not attempting in the Diagnostics json like having to sibling nodes in the replica list one for attempting other for not attempting - that way you can keep this mutable state out of the precalculated Uri string. Also easier to read IMO.

Reply:
Changed the replicaStatusList to be in this format 
"replicaStatusList": { 
  "Attempting": ["25475:Primary:Connected"],
  "NotAttempting": ["nnnnn:Secondary:UnhealthyPending"]
},